pkg/k8scfgmap/watch.go

```diff
@@ -0,0 +1,87 @@
+package k8scfgmap
+
+import (
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	k8sTest "k8s.io/client-go/testing"
+)
+
+var log = logging.LoggerForModule()
+
+// ConfigMapWatcher watches a config map in a given namespaces and evokes a callback function
+// when changes are detected.
+type ConfigMapWatcher struct {
```

Comment:
I'm a bit torn whether the watcher itself should be a struct or simply a function that starts it already. Right now, while you could chain it like `NewConfigMapWatcher(...).Watch()` we could also do something like `WatchConfigMap(...)`. It doesn't necessarily needs to be an exported struct, since we are only accessing one method externally.

Reply:
True, but exposing the struct also does not seem like a big deal. And it would make it easier if we ever decide to add more methods to the watcher. So I would prefer to leave it as is.